File: fix-callback/Send.java
import java.util.concurrent.CompletableFuture;

public class Send {
    public CompletableFuture<Object> newSend(Object request) {
      // Create a CompletableFuture to hold the result
      CompletableFuture<Object> future = new CompletableFuture<>();

      // Call the send() function and provide a callback to set the result
      _send(request, (Object result) -> {
        future.complete(result);
      });

      // Return the CompletableFuture
      return future;
    }
}

File: fix-callback/send.dart
import 'dart:async';

Future<dynamic> send(dynamic request) async {
  // Create a Completer to hold the result
  Completer<dynamic> completer = Completer<dynamic>();

  // Call the send() function and provide a callback to set the result
  _send(request, (dynamic result) {
    completer.complete(result);
  });

  // Return the Future for the result
  return completer.future;
}

